### PR TITLE
doc: release notes update for 10.2.10

### PR DESCRIPTION
**Pull request by theanalyst**
Added the release notes for v10.2.10 & updated the releases table.

### CODE REVIEW

File: doc/release-notes.rst [hunk 1]
```diff
@@ -5930,6 +5930,129 @@ Notable Changes
...
+* rgw: fix infinite loop in rest api for log list (`issue#20386 <http://tracker.ceph.com/issues/20386>`_, `pr#15988 <https://github.com/ceph/ceph/pull/15988>`_, xierui, Casey Bodley)
+* rgw: fix race in RGWCompleteMultipart (`issue#20861 <http://tracker.ceph.com/issues/20861>`_, `pr#16767 <https://github.com/ceph/ceph/pull/16767>`_, Abhishek Varshney, Matt Benjamin)
+* rgw: Fix up to 1000 entries at a time in check_bad_index_multipart (`issue#20772 <http://tracker.ceph.com/issues/20772>`_, `pr#16880 <https://github.com/ceph/ceph/pull/16880>`_, Orit Wasserman, Matt Benjamin)
+* rgw: folders starting with _ underscore are not in bucket index (`issue#19432 <http://tracker.ceph.com/issues/19432>`_, `pr#16276 <https://github.com/ceph/ceph/pull/16276>`_, Giovani Rinaldi, Orit Wasserman)
+* rgw: 'gc list --include-all' command infinite loop the first 1000 items (`issue#19978 <http://tracker.ceph.com/issues/19978>`_, `pr#15719 <https://github.com/ceph/ceph/pull/15719>`_, Shasha Lu, fang yuxiang)
+* rgw: meta sync thread crash at RGWMetaSyncShardCR (`issue#20251 <http://tracker.ceph.com/issues/20251>`_, `pr#16711 <https://github.com/ceph/ceph/pull/16711>`_, Nathan Cutler)
```
Comment (smithfarm) [inline doc/release-notes.rst]: please s/Nathan Cutler/fang yuxiang, Nathan Cutler/

File: doc/release-notes.rst [hunk 2]
```diff
@@ -5930,6 +5930,129 @@ Notable Changes
...
+* rgw: multipart copy-part remove '/' for s3 java sdk request header (`issue#20075 <http://tracker.ceph.com/issues/20075>`_, `pr#16266 <https://github.com/ceph/ceph/pull/16266>`_, root)
```
Comment (smithfarm) [inline doc/release-notes.rst]: please s/root/donglinpeng/

**Comment by smithfarm** [comment]
Weird, https://github.com/ceph/ceph/pull/18117 was merged but this PR is not showing the merge conflict.

**Comment by theanalyst** [comment]
@smithfarm I recently  rebased to fix the conflict :)

**Comment by smithfarm** [comment]
@theanalyst Yup. I'd say this is good to go.

**Pull request merged**